src/main/java/com/google/cloud/spanner/r2dbc/SpannerConnectionFactoryProvider.java

```
@@ -34,9 +38,25 @@
   /** R2DBC driver name for Google Cloud Spanner. */
   public static final String DRIVER_NAME = "spanner";
 
+  /** Option name for GCP Project. */
+  public static final Option<String> OPTION_PROJECT = Option.valueOf("project");
+
+  /** Option name for GCP Spanner instance. */
+  public static final Option<String> OPTION_INSTANCE = Option.valueOf("instance");
+
   @Override
   public ConnectionFactory create(ConnectionFactoryOptions connectionFactoryOptions) {
-    return new SpannerConnectionFactory(null);
+    SpannerConnectionConfiguration config = new SpannerConnectionConfiguration.Builder()
+        .setProjectId(connectionFactoryOptions.getValue(OPTION_PROJECT))
```

That should be `getRequiredValue(…)` to fail if a value is absent.